File: src/sdk-client/Microsoft.Windows.Midi.Native/MidiDeviceInformation.cpp
// ------------------------------------------------------------------------------------------------
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See LICENSE in the GitHub project root for license information.
// ------------------------------------------------------------------------------------------------

#define WINDOWSMIDISERVICES_EXPORTS

#include <string>
#include <filesystem>
#include <functional>
#include <map>

#include "WindowsMidiServicesSession.h"
#include "WindowsMidiServicesEnumeration.h"

namespace Microsoft::Windows::Midi::Enumeration
{

	struct MidiDeviceInformation::implMidiDeviceInformation
	{
	public:
		MidiObjectId Id;
		MidiObjectId TransportId;
		std::unique_ptr<std::wstring> Name;
		std::unique_ptr<std::wstring> DeviceSuppliedName;
		std::unique_ptr<std::string>  Serial;
		std::unique_ptr<std::wstring> IconFileName;
		std::unique_ptr<std::wstring> Description;

		implMidiDeviceInformation(MidiObjectId id, MidiObjectId transportId)
		{
			Id = id;
			TransportId = transportId;
		}
	};

	MidiDeviceInformation::MidiDeviceInformation(MidiObjectId id, MidiObjectId transportId)
	{
		_pimpl = new implMidiDeviceInformation(id, transportId);
	}

	MidiDeviceInformation::~MidiDeviceInformation()
	{
		delete _pimpl;
	}

	MidiDeviceInformation::MidiDeviceInformation(const MidiDeviceInformation& info)
	{
		_pimpl = nullptr;
	}


	MidiObjectId MidiDeviceInformation::getId()
	{
		return MidiObjectId{};
	}

	MidiObjectId MidiDeviceInformation::getTransportId()
	{
		return MidiObjectId{};
	}

	const wchar_t* MidiDeviceInformation::getName()
	{
		return nullptr;
	}

	const wchar_t* MidiDeviceInformation::getDeviceSuppliedName()
	{
		return nullptr;
	}

	const char8_t* MidiDeviceInformation::getSerial()
	{
		return nullptr;
	}

	const wchar_t* MidiDeviceInformation::getIconFileName()
	{
		return nullptr;
	}

	const wchar_t* MidiDeviceInformation::getDescription()
	{
		return nullptr;
	}

	const bool MidiDeviceInformation::getIsRuntimeCreated()
	{
		return false;
	}

	const uint32_t MidiDeviceInformation::getOwningProcessIdIfRuntimeCreated()
	{
		return 0;
	}


}

File: src/sdk-client/zzold/Original CS Prototype/Windows.Midi/Enumeration/MidiDeviceInformation.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See LICENSE in the project root for license information.

using Microsoft.Windows.Midi.Internal.Data;

using Windows.Foundation;

namespace Microsoft.Windows.Midi.Enumeration
{
    public sealed class MidiDeviceInformation
    {
        private MidiDeviceSerializableData _deviceData;

        /// <summary>
        /// Unique Id for this device
        /// </summary>
        public Guid Id { get => _deviceData.Id; }

        /// <summary>
        /// The transport type used to speak to this device
        /// </summary>
        public Guid DevicePluginId { get => _deviceData.DevicePluginId; }

        /// <summary>
        /// User-supplied name for this specific device
        /// </summary>
        public string Name { get => _deviceData.Name; }

        /// <summary>
        /// Device or driver-supplied name for this device
        /// </summary>
        public string DeviceSuppliedName { get => _deviceData.DeviceSuppliedName; }

        /// <summary>
        /// If the device has a unique serial number, it is here
        /// </summary>
        public string Serial { get => _deviceData.Serial; }

        /// <summary>
        /// File name for the icon for this device
        /// </summary>
        public string IconFileName { get => _deviceData.IconFileName; }

        /// <summary>
        /// Description of the device
        /// </summary>
        public string Description { get => _deviceData.Description; }



    }
}
